SwanContents/swancontents/serverextension/templates/page.html

```diff
@@ -0,0 +1,92 @@
+<!DOCTYPE HTML>
+<html>
+
+<head>
+
+    <meta charset="utf-8">
+
+    <title>{% block title %}SWAN{% endblock %}</title>
+    {% block favicon %}<link id="favicon" rel="shortcut icon" type="image/x-icon" href="{{ static_url("favicon.ico") }}">
+    {% endblock %}
+    <link rel="stylesheet" href="{{static_url("style/bootstrap.min.css") }}" />
+    <link rel="stylesheet" href="{{static_url("style/bootstrap-theme.min.css") }}" />
+    <link rel="stylesheet" href="{{static_url("style/index.css") }}" />
+    <meta http-equiv="X-UA-Compatible" content="IE=edge" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+
+    {% block stylesheet %}
+    {% endblock stylesheet %}
+
+    {% block meta %}
+    {% endblock meta %}
+
+</head>
+
+<body class="{% block bodyclasses %}{% endblock %}" {% block params %} {% if logged_in and token %}
+  data-jupyter-api-token="{{token | urlencode}}" {% endif %} {% endblock params %} dir="ltr">
+
+  <noscript>
+    <div id='noscript'>
+      {% trans %}Jupyter Server requires JavaScript.{% endtrans %}<br>
+      {% trans %}Please enable it to proceed. {% endtrans %}
+    </div>
+  </noscript>
+  <div id="header" role="navigation" aria-label="{% trans %}Top Menu{% endtrans %}">
+    <div id="header-container" class="container">
+      <div id="jupyter_server" class="nav navbar-brand"><a href="{{default_url}}
+    {%- if logged_in and token -%}?token={{token}}{%- endif -%}" title='{% trans %}dashboard{% endtrans %}'>
+          {% block logo %}<img src='{{static_url("logo/logo.png") }}' alt='SWAN' />{% endblock %}
+        </a></div>
+
+      {% block headercontainer %}
+      {% endblock headercontainer %}
+
+      {% block header_buttons %}
+      {% endblock header_buttons %}
+
+    </div>
+    <div class="header-bar"></div>
+
+    {% block header %}
+    {% endblock header %}
+  </div>
+
+  <div id="site">
+    {% block site %}
+    {% endblock site %}
+  </div>
+
+  {% block after_site %}
+  {% endblock after_site %}
+
+  {% block script %}
+  {% endblock script %}
+
+  <script type='text/javascript'>
+    function _remove_token_from_url() {
```

Comment:
I don't know if we discussed about this function? What does it do? Some doc would be nice?

Reply:
We discussed. It comes from upstream, it's part of aligning our files with upstream changes

Reply:
True!

Reply:
Updated the commit message to state that the update means pulling stuff from upstream.